packages/react-native/ReactAndroid/cmake-utils/ReactNative-application.cmake

```diff
@@ -39,9 +39,16 @@ if (PROJECT_ROOT_DIR)
 # variable is defined if user need to access it.
 endif ()
 
-file(GLOB input_SRC CONFIGURE_DEPENDS
-        ${REACT_ANDROID_DIR}/cmake-utils/default-app-setup/*.cpp
-        ${BUILD_DIR}/generated/autolinking/src/main/jni/*.cpp)
+file(GLOB override_cpp_SRC CONFIGURE_DEPENDS *.cpp)
+if(override_cpp_SRC)
```

Comment:
```suggestion
# We check if the user is providing a custom OnLoad.cpp file. If so, we pick that
# for compilation. Otherwise we fallback to using the `default-app-setup/OnLoad.cpp` 
# file instead.
if(override_cpp_SRC)
```